src/java.base/share/classes/java/util/concurrent/StructuredTaskScope.java

```diff
@@ -0,0 +1,1281 @@
+            super.handleComplete(subtask);
+
+            if (firstResult != null) {
+                // already captured a result
+                return;
+            }
+
+            if (subtask.state() == Subtask.State.SUCCESS) {
+                // task succeeded
+                T result = subtask.get();
+                Object r = (result != null) ? result : RESULT_NULL;
+                if (FIRST_RESULT.compareAndSet(this, null, r)) {
+                    super.shutdown();
+                }
+            } else if (firstException == null) {
+                // capture the exception thrown by the first subtask that failed
+                FIRST_EXCEPTION.compareAndSet(this, null, subtask.exception());
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         * @return this task scope
+         * @throws IllegalStateException {@inheritDoc}
+         * @throws WrongThreadException {@inheritDoc}
+         */
+        @Override
+        public ShutdownOnSuccess<T> join() throws InterruptedException {
+            super.join();
+            return this;
+        }
+
+        /**
+         * {@inheritDoc}
+         * @return this task scope
+         * @throws IllegalStateException {@inheritDoc}
+         * @throws WrongThreadException {@inheritDoc}
+         */
+        @Override
+        public ShutdownOnSuccess<T> joinUntil(Instant deadline)
+            throws InterruptedException, TimeoutException
+        {
+            super.joinUntil(deadline);
+            return this;
+        }
+
+        /**
+         * {@return the result of the first subtask that completed {@linkplain
+         * Subtask.State#SUCCESS successfully}}
+         *
+         * <p> When no subtask completed successfully, but a subtask {@linkplain
+         * Subtask.State#FAILED failed} then {@code ExecutionException} is thrown with
+         * the subtask's exception as the {@linkplain Throwable#getCause() cause}.
+         *
+         * @throws ExecutionException if no subtasks completed successfully but at least
+         * one subtask failed
+         * @throws IllegalStateException if the handleComplete method was not invoked with
+         * a completed subtask, or the task scope owner did not invoke join after forking
+         * @throws WrongThreadException if the current thread is not the task scope owner
+         */
+        public T result() throws ExecutionException {
+            ensureOwnerAndJoined();
+
+            Object result = firstResult;
+            if (result == RESULT_NULL) {
+                return null;
+            } else if (result != null) {
+                @SuppressWarnings("unchecked")
+                T r = (T) result;
+                return r;
+            }
+
+            Throwable ex = firstException;
+            if (ex != null) {
+                throw new ExecutionException(ex);
+            }
+
+            throw new IllegalStateException("No completed subtasks");
```

Comment:
I believe it may be possible to implement as the following if you so wish:
```suggestion
            return result(ExecutionException::new);
```

Reply:
Good, avoids duplication.